pkg/crd/markers/package.go

```diff
@@ -22,7 +22,7 @@ import (
 
 func init() {
 	AllDefinitions = append(AllDefinitions,
-		must(markers.MakeDefinition("groupName", markers.DescribesPackage, "")).
+		mustOptional(markers.MakeDefinition("groupName", markers.DescribesPackage, "")).
```

Comment:
Is making this optional enough to handle groupName= correctly? (or will we e.g. just ignore the marker?)

(Sorry stupid question, but not really familiar with this)

Reply:
> Is making this optional enough to handle groupName= correctly? (or will we e.g. just ignore the marker?)

It was at least enough to make it generate CRDs again with the expected outcome after an upgrade to 1.31.0-alpha.3.
 
> (Sorry stupid question, but not really familiar with this)

Me neither, trial and error ftw!!1 :sweat_smile:

Reply:
> > Is making this optional enough to handle groupName= correctly? (or will we e.g. just ignore the marker?)
> 
> It was at least enough to make it generate CRDs again with the expected outcome after an upgrade to 1.31.0-alpha.3.

That said, it was generating the expected outcome for _me_, i.e. CRDs holding core/v1 structs could be generated again. The errors I saw w/o this look like "unable to locate schema for type "k8s.io/api/core/v1".PullPolicy" ...  Not sure what "handle groupName= correctly" entails from your point of view, though.

Reply:
> entails from your point of view, though.

Yeah, same :)

Reply:
I'm going to test this with Cluster API. We have a lot of CRDs in there. I'll check if there is a relevant diff after this PR

Reply:
Tested with CAPI, no unexpected diffs